File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/internal/Work.java
package ru.bozaro.gitlfs.client.internal;

import ru.bozaro.gitlfs.common.data.Link;

import javax.annotation.Nonnull;
import java.io.IOException;

/**
 * Work with same auth data.
 *
 * @author Artem V. Navrotskiy
 */
public interface Work<R> {
  R exec(@Nonnull Link auth) throws IOException;
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/internal/Work.kt
package ru.bozaro.gitlfs.client.internal

import ru.bozaro.gitlfs.common.data.Link
import java.io.IOException

/**
 * Work with same auth ru.bozaro.gitlfs.common.data.
 *
 * @author Artem V. Navrotskiy
 */
fun interface Work<R> {
    @Throws(IOException::class)
    fun exec(auth: Link): R
}
